node/test/util/commit.js

```diff
@@ -3449,12 +3449,20 @@ describe("commit mid-merge", function() {
         const index = yield sRepo.index();
         const sEntry = index.getByIndex(1);
         const newEntry = new NodeGit.IndexEntry();
-        newEntry.flags = sEntry.flags;
+        // mask off the stage bits here, so that the new entry we write is
+        // in stage zero.
+        newEntry.flags = sEntry.flags & ~0x3000;
```

Comment:
can you explain this one to me?

Reply:
We're trying, here, to mark a conflict as resolved (never mind that there are still conflict markers -- this is just a test).  Once a conflict is resolved, the stage should be 0.   The stage bits are 0x3000.  We want everything *but* the stage bits, so we bitwise-and with the bitwise-inverse of 0x3000.  So imagine the original bits are 0x6abc.  0x6abc & 0xcfff gives 0x4abc  -- that is, everything but the stage bits are the same, and the stage bits are zero.  So now the entry's stage is zero, which I think is what `git add` does.